test/test_hf_tokenizer.cpp

```diff
@@ -56,7 +56,7 @@ TEST(HFTokenizerTest, TestEncode) {
   // "Hello world!" should tokenize to something like [1, 8, 9] or [1, 4, 5, 6,
   // 7] depending on how the BPE merges work
   EXPECT_GT(result.get().size(), 0);
-  EXPECT_EQ(result.get()[0], 2); // BOS token (ID 2 in our test tokenizer)
+  EXPECT_EQ(result.get()[0], 0); // BOS token (ID 0 in our test tokenizer)
```

Comment:
I think 0 is just the default value for the bos token, I don't see any bos tokens in the .json https://github.com/pytorch-labs/tokenizers/blob/e894ef40ff57f61e40f76c69ca0ee3abe674cf6b/include/pytorch/tokenizers/tokenizer.h#L71

Also can follow up on this but I'm not quite sure what this test is really testing for